### ISSUE TITLE
exercism v3 track 

### ISSUE THREAD

**Issue by RoelofWobben** [exercism/pharo-smalltalk#413]
Helllo, 

I asked here for some help to see if and how Pharo can be in exercism v3. 

https://github.com/exercism/v3/issues/1909

Roelof

**Comment by RoelofWobben** [comment]
Answer from the exercism co founder : 

In-browser coding isn't a compulsory requirement, and for Phara, from what I understand, it's an unachievable goal. So we'll have to do without.

So the main priority would starting to be building out Concept Exercises.

**Comment by RoelofWobben** [comment]
Maybe a seperate issue to discuss some concepts we want to learn our users. 
I was thinking about streams, class v instance variables

**Comment by RoelofWobben** [comment]
I thinking of these subjects in this order 

- strings ( There is a exercise we can convert) 
- numbers ( There is also a exercise  for it we can use) 
 -  if then ( also a existing exercise) 
-  exercise where we learn a user collect and reject 
-  streams 
-  double dispatch

**Comment by samWson** [comment]
@RoelofWobben thanks for getting this started. At this time it doesn't look like we need to change the high level details of how we are running this track. That's a good thing and something we don't have to worry about. For now I'm going to focus on making sure people can contribute to the track and make it easier to maintain this repo.

**Comment by samWson** [comment]
@RoelofWobben have you read [Pharo by Example](http://books.pharo.org/updated-pharo-by-example/pdf/2018-09-29-UpdatedPharoByExample.pdf)? It was the book that helped me understand some of Pharos core idioms and the object model. It could be useful for some inspiration on concept exercises.

In particular the chapter on [Streams](http://books.pharo.org/updated-pharo-by-example/pdf/2018-09-29-UpdatedPharoByExample.pdf#chapter.11) might be useful since they are used for collections and file access. Also [The Pharo Object Model](http://books.pharo.org/updated-pharo-by-example/pdf/2018-09-29-UpdatedPharoByExample.pdf#chapter.6) is pretty useful for understanding Pharo. 

Other chapters like Morphic might not be useful since its being replaced by Bric and Blok in Pharo.

**Comment by RoelofWobben** [comment]
I did not read it .  I did the Mooc and as first book I have read this one:  https://books.pharo.org/learning-oop/
But I think I can read this book and see if we can make a challenge of it. 

My plan if you agree is too port the strings and numbers concept exercises to Pharo and before I commit them to exercism discuss if I explain things right.

**Comment by RoelofWobben** [comment]
What do you think of this as first exercise. Lot;s of languges of the new track uses this as first exercise. 
Is this well written smalltalk or can I better change things so the user learns it the right way 

https://www.dropbox.com/t/SQJ5wrELVH4dkEAv

**Comment by samWson** [comment]
It looks like a simple exercise in arithmetic which makes it a good introduction to TDD.

You could shorten `Lasagna new totalTimeInMinutes: 4 timeInOven: 8` to `Lasagna new totalMinutes: 4 ovenTime: 8`. I haven't read the exercise background so I don't know what `totalTimeInMinutes` is representing? Is it total preparation time?

In any case this is a good start. Finish this exercise if you want but you don't need to make much more for the moment. We also need to know how the v3 exercises are going to fit in the repository, what the file structure of each exercise is etc. It will be useful having a small number of v3 exercises around to experiment with. Once we know more then we can start making more exercises.

**Comment by RoelofWobben** [comment]
yep, It is a good start but like I try to say we have to rewrite the generator so it outputs the exercise in the new format. 

The new format is known.. 

.docs 
---   after.md   
---   hints,md 
--   instructions.md 
--  introduction,md 
 .meta 
-- example.st 
-- config,json 
-- design.md 
exercise.st 
exercise_test.st 

so in the root there are the folders .meta ,.docs andt the exercise file and the test file
